specification/BUILDING.md

```diff
@@ -0,0 +1,110 @@
+- [Cross-linking between files](#cross-linking-between-files)
+
+## Building the specification 
+
+The following is a short summary of the basic process for generating a single HTML- or PDF file containing the whole specification.
+
+> Note: This process might be extended in the future, with details about how [wetzel](https://github.com/CesiumGS/wetzel) is used to generate the property reference. Depending on the exact process, the required toolchain might be summarized in a Docker container, similar to [Vulkan](https://github.com/KhronosGroup/Vulkan-Docs/blob/15d807ce4839d8feb523ca5c133a42a2aa448ade/BUILD.adoc), and controlled via a Makefile.
+
+### Asciidoctor setup
+
+- Install the Ruby interpreter, 2.3 or later, from http://www.ruby-lang.org/
+- Install Asciidoctor: `gem install asciidoctor`
+- In order to be able to generate PDF output: `gem install asciidoctor-pdf`
+- Install some rogue software: `gem install rouge` - no worries, that's the syntax highlighter...
+
+- A VSCode plugin for AsciiDoc syntax highlighting and preview: https://marketplace.visualstudio.com/items?itemName=asciidoctor.asciidoctor-vscode
+
+### Generating HTML and PDF with AsciiDoc
+
+- Generating HTML:
+  - `asciidoctor --verbose Specification.adoc --out-file Specification-1.1.0.html`
+- Generating PDF:
+  - `asciidoctor-pdf --verbose --trace --attribute scripts=cjk --attribute pdf-theme=default-with-fallback-font Specification.adoc --out-file Specification-1.1.0.pdf`
+  
+    > Note: The last call is derived from https://asciidoctor.org/docs/asciidoctor-pdf/#support-for-non-latin-languages and handles Japanese characters. Without them, the call could just be `asciidoctor -r asciidoctor-pdf -b pdf Specification.adoc -o Specification-1.1.0.pdf`
+
+<sup>If the call does not appear to do anything, neither generate a PDF nor print an error message, make sure you typed `asciidoctor` and not just `asciidoc`</sup>
+
+
+### Compressing the PDF
+
+By default, the PDF file is large, due to the large image files being inserted uncompressed. There are tools for compressing the PDF, including a dedicated `asciidoctor-pdf-optimize` tool, but they have caveats. 
+
+The process that worked for me:
+
+- Download GhostScript **not later than 9.20** from https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/tag/gs920  
+  - This is the latest version that did not contain the so-called ""bugfix"" from https://bugs.ghostscript.com/show_bug.cgi?id=699830 that apparently caused more harm than good.
+     
+  - Run the following command: 
+    ```
+    "C:\Program Files\gs\gs9.20\bin\gswin64.exe" ^
+      -dPrinted=false ^
+      -sDEVICE=pdfwrite ^
+      -dCompatibilityLevel=1.4 ^
+      -dPDFSETTINGS=/screen ^
+      -dNOPAUSE ^
+      -dBATCH ^
+      -dDetectDuplicateImages ^
+      -sOutputFile=Specification-1.1.0-compressed.pdf ^
+      Specification-1.1.0.pdf   
+    ```
+    (This is a Windows .BAT file. On Linux, replace the `^` with `\`)
```

Comment:
My notes on the build process on Linux:

* I had to build Ghostscript from source since I couldn't find precompiled Linux binaries. I downloaded `ghostscript-9.20-linux-x86_64.tgz`, ran `./configure`, then `make`, and that created `bin/gs`.
* Have you seen this warning? I'm not sure which character it's refering to.

```
~/Code/3d-tiles/specification$ asciidoctor-pdf --verbose --trace --attribute scripts=cjk --attribute pdf-theme=default-with-fallback-font Specification.adoc --out-file Specification-1.1.0.pdf
asciidoctor: WARNING: Could not locate the character `⁣' (\u2063) in the following fonts: Noto Serif, M+ 1p Fallback, Noto Emoji
```

Reply:
I saw that warning, and tried to figure out where it came from, but my friggin Windows console just displayed this as "�" and did not include the info that it was `\u2063`, so I couldn't properly search for this. I'll take a note for that as well, the Unicode code may be helpful to figure out what this actually is... "Invisible Separator" for now, but I'll have a look...